event/src/unvalidated.rs

```diff
@@ -0,0 +1,249 @@
+//! Types of raw unvalidated Ceramic Events
+
+use std::collections::BTreeMap;
+
+use cid::Cid;
+use ipld_core::ipld::Ipld;
+use serde::{Deserialize, Serialize};
+
+use crate::bytes::Bytes;
+
+/// Ceramic Event
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Event {
+    /// Time event in a stream
+    Time(Box<TimeEvent>),
```

Comment:
why do TimeEvents need to be boxed but not SignedEvents/JWS?

Reply:
TimeEvent is large ~312 bytes and JWS is small < 8 bytes. So putting TimeEvent in a Box allows both variants to be about the same size making the enum more efficient. Honestly this is a bit of a premature optimization but clippy complained so I didn't fight it.

Reply:
except the JWS payload can be up to 256kb so there's no guarantee the JWS is actually small either.

At the least can we add some comments explaining the reasoning of whatever we decide to go with?

Reply:
Yes but those bytes will be allocated on the heap via the Vec<u8> of the payload. The JWS struct is basically just two pointers.

Reply:
In contrast the TimeEvent struct has three CIDs which are 64+ bytes each plus the string bits. So that struct is ~312 bytes (according to the Rust compiler). I'll add a comment.